src/components/common/Modal/index.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { ReactNode, FC } from 'react';
+import {
+  DialogOverlay as ReachDialogOverlay,
+  DialogContent as ReachDialogContent,
+} from '@reach/dialog';
+import { pick, omit } from '@styled-system/props';
+import { motion, AnimatePresence } from 'framer-motion';
+import { Box, BoxProps, Text } from 'rebass/styled-components';
+import styled from 'styled-components';
+import { Icon } from '~components';
+import { Close } from '~components/common/Icon/library';
+
+const MotionOverlay = motion.custom(ReachDialogOverlay);
+const MotionDialog = motion.custom(ReachDialogContent);
+
+const DialogOverlay = styled(MotionOverlay)`
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  z-index: ${(p) => p.theme.zIndices.modal};
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding: ${(p) => p.theme.space[4]}px;
+  overflow: auto;
+  background: rgba(0, 0, 0, 0.4);
+  opacity: 0.75;
+
+  @media (max-width: ${(p) => p.theme.breakpoints.md}) {
+    padding: ${(p) => p.theme.space[4]}px ${(p) => p.theme.space[2]}px 92px;
+  }
+`;
+
+const DialogContent = styled(MotionDialog)<BoxProps>`
+  position: relative;
+  flex: 1;
+  padding: 0;
+  margin: 0;
+  max-height: 100%;
+  background: transparent;
+
+  @media (min-width: ${(p) => p.theme.breakpoints.md}) {
+    max-width: 774px;
+  }
+`;
+
+const CloseButton = styled.button`
+  position: absolute;
+  right: 16px;
+  top: 16px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 24px;
+  height: 24px;
+  transition: color 0.2s;
+
+  &:hover {
+    svg {
+      color: ${(p) => p.theme.colors.grey800};
+    }
+  }
+`;
+
+const ContentWrapper = styled(motion.div)`
+  position: relative;
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  max-height: calc(100vh - 160px);
+  margin: 64px 0;
+  padding: ${(p) => p.theme.space[5]}px;
+  overflow: hidden;
+  background: ${(p) => p.theme.colors.white};
+  outline: none;
+  border-radius: ${(p) => p.theme.radii.sm};
+  box-shadow: ${(p) => p.theme.shadows.sm};
+`;
+
+type ModalProps = {
+  isOpen: boolean;
+  close(any): void;
+  ariaLabel?: string;
+  header?: ReactNode;
+  footer?: ReactNode;
+  children?: ReactNode;
+};
+
+export const ModalBase: FC<ModalProps & BoxProps> = ({
+  children,
+  ariaLabel,
+  isOpen,
+  close,
+}) => {
+  return (
+    <AnimatePresence>
+      {isOpen && (
```

Comment:
I'm not sure if we should use the flag to render the modal. if the modal get's expensive it might be a good way to go since it won't be rendered until triggered. But I think passing `isOpen` to the `dialogOverlay` should be the way to go.

Reply:
for some reason that prop doesn't work in combination with the `AnimatePresence`